From 08d624dabe4fb489d56bbff45014c78612f4a5c9 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Wed, 3 Jan 2018 09:57:29 +0000
Subject: [PATCH] Fix systemd-resolved failures

Port 2 commits from https://github.com/fedora-selinux/selinux-policy:
  d1f068d3 Allow systemd-resolved bind to dns port. BZ(1400023)
  5bf29bd0 Allow systemd_resolved to connect on system bus. BZ(1366334)

Adjust rules according to Fedora 25 runtime status:
  +init_write_pid_socket(systemd_resolved_t)
  +init_dgram_send(systemd_resolved_t)

Remove error:
  Failed to start Network Name Resolution

Fix avc denials:
  avc: denied { net_raw } for pid=333 comm="systemd-resolve" \
  scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tclass=capability

  avc: denied { name_bind } for pid=333 comm="systemd-resolve" src=53 \
  scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:dns_port_t:s0 tclass=udp_socket

  avc: denied { acquire_svc } for service=org.freedesktop.resolve1 \
  scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:system_dbusd_t:s0-s15:c0.c1023 tclass=dbus

  avc: denied { write } for pid=677 comm="systemd-resolve" \
  name="notify" dev="tmpfs" ino=9416 \
  scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:init_var_run_t:s0 tclass=sock_file

  avc: denied { sendto } for pid=345 comm="systemd-resolve" \
  path="/run/systemd/notify" \
  scontext=system_u:system_r:systemd_resolved_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 \
  tclass=unix_dgram_socket

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index e24c308..728eea1 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1015,6 +1015,8 @@ seutil_read_file_contexts(systemd_resolved_t)
 systemd_log_parse_environment(systemd_resolved_t)
 systemd_read_networkd_runtime(systemd_resolved_t)
 
+init_write_pid_socket(systemd_resolved_t)
+
 optional_policy(`
 	dbus_connect_system_bus(systemd_resolved_t)
 	dbus_system_bus_client(systemd_resolved_t)
-- 
2.7.4

